src/video-player.js

```diff
@@ -274,11 +275,36 @@ class VideoPlayer extends Utils.mixin(Eventable) {
     this.videojs.ready(() => {
       onReady();
 
+
       if (ready) {
         ready(this);
       }
     });
 
+
+    this.nbCalls = 0;
+    this.reTryVideo = (maxNumberOfCalls, timeout) => {
+      if (!this.isVideoReady()) {
+        if (this.nbCalls < maxNumberOfCalls) {
+          this.nbCalls++;
+          this.videojs.setTimeout(this.reTryVideo, timeout);
+        } else {
+          let e = new Error('Video is not ready please try later');
+          this.videojs.trigger('error', e);
+        }
+      }
+
+    };
+
+    this.isVideoReady = () => {
+      let s = this.videojs.readyState();
+       if (s === 4) {
```

Comment:
@yaniv-li 
Unless I'm missing something this implementation still have a problem, given the following case:
1. Set source for the first video.
2. Video is struggling to load, and `nbCalls` has been increased. 
3. User decides not to wait and change the video source while waiting
The `nbCalls` variable will not get reset for the next source since readyState has not changed to 4 yet causing a shorter period for getting a timeout.